### PR TITLE
Logging download failure as telemetry from RM extension

### PR DESCRIPTION
**Pull request by GitHubSriramB**

### CODE REVIEW

File: src/Agent.Worker/Release/ReleaseJobExtension.cs [hunk 1]
```diff
@@ -102,14 +110,25 @@ public ReleaseJobExtension()
...
+                }
+            }
+            catch (Exception ex)
             {
-                // TODO: Create this as new task. Old windows agent does this. First is initialize which does the above and download task will be added based on skipDownloadArtifact option
-                executionContext.Output("Downloading artifact");
+                Thread.Sleep(15 * 1000);
 
-                await DownloadArtifacts(executionContext, teamProjectId, artifactsWorkingFolder, releaseId);
+                HostContext.GetService<ICommandHandler>().TryProcessCommand(executionContext, GetTelemetryMessage(ex));
```
Comment (TingluoHuang) [inline src/Agent.Worker/Release/ReleaseJobExtension.cs]: you have executionContext, so just call executionContext.addissue().

File: src/Agent.Worker/Release/ReleaseJobExtension.cs [hunk 2]
```diff
@@ -102,14 +110,25 @@ public ReleaseJobExtension()
...
+                    executionContext.Output("Downloading artifact");
+
+                    await DownloadArtifacts(executionContext, teamProjectId, artifactsWorkingFolder, releaseId);
+                }
+            }
+            catch (Exception ex)
             {
-                // TODO: Create this as new task. Old windows agent does this. First is initialize which does the above and download task will be added based on skipDownloadArtifact option
-                executionContext.Output("Downloading artifact");
+                Thread.Sleep(15 * 1000);
```
Comment (TingluoHuang) [inline src/Agent.Worker/Release/ReleaseJobExtension.cs]: why need sleep?

Comment (GitHubSriramB) [inline src/Agent.Worker/Release/ReleaseJobExtension.cs]: oops, missed removing it post testing. will revert.

File: src/Agent.Worker/Release/ReleaseJobExtension.cs (hunk 1 above)
Comment (GitHubSriramB) [inline src/Agent.Worker/Release/ReleaseJobExtension.cs]: sure.

**Comment by TingluoHuang** [comment]
@GitHubSriramB Looks good.

**Pull request merged**